### PR TITLE
[X86_64] fix SSE type error in vaarg.

### LINKED ISSUE #86371: [X86_64] va_arg with SSE type get error
demo.c
```c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>

struct S271 { union{}a; double b;} ;
struct S271 s271;
void check271va (int z, ...) {
    struct S271 arg;
    va_list ap;
    __builtin_va_start(ap, z);
    arg = __builtin_va_arg(ap, struct S271);
    long double a = __builtin_va_arg(ap, long double);
    printf("%Lf\n", a);
    if (a != 2.0L)
       printf("Fail\n");
    __builtin_va_end(ap);
}
int main (void) {
    printf("%lu\n", sizeof(s271));
    check271va (2, s271, 2.0L);
}
```
clang
```
16
0.000000
Fail
```
gcc
```
16
2.000000
```
https://godbolt.org/z/o6v318vWM

### PR DESCRIPTION
**Pull request by CoTinker**
tweak the position of the ++neededSSE when Lo is NoClass and Hi is SSE. Fix #86371.

### CODE REVIEW

File: clang/test/CodeGenCXX/x86_64-vaarg.cpp [hunk 1]
```diff
@@ -21,3 +21,18 @@ empty empty_record_test(int z, ...) {
   __builtin_va_start(list, z);
   return __builtin_va_arg(list, empty);
 }
+
```
Comment (efriedma-quic) [inline clang/test/CodeGenCXX/x86_64-vaarg.cpp:24]: Please regenerate the CHECK lines with update_cc_test_checks.py.

Comment (CoTinker) [reply clang/test/CodeGenCXX/x86_64-vaarg.cpp:24]: done

File: clang/test/CodeGen/X86/x86_64-vaarg.c [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+empty empty_record_test(int z, ...) {
+  __builtin_va_list list;
+  __builtin_va_start(list, z);
+  return __builtin_va_arg(list, empty);
+}
+
+typedef struct {
+  struct{} a;
+  double b;
+} s1;
```
Comment (efriedma-quic) [inline clang/test/CodeGen/X86/x86_64-vaarg.c:30]: Not sure this is really worth testing... at least, it's not testing anything relevant to this patch.  (sizeof(s1)==8.)  Up to you, I guess.

**Review by efriedma-quic — APPROVED** [review]
LGTM

**Comment by CoTinker** [comment]
Thanks.

**Comment by AtariDreams** [comment]
/cherry-pick https://github.com/llvm/llvm-project/commit/9c8dd5e6f6bd93deb95de9642632223f54a18a11

**Pull request merged**